addons/sale/models/sale_order.py

```diff
@@ -793,6 +793,12 @@ def _unlink_except_draft_or_cancel(self):
                     "You can not delete a sent quotation or a confirmed sales order."
                     " You must first cancel it."))
 
+    def write(self, vals):
+        res = super().write(vals)
+        if 'partner_id' in vals and vals.get('partner_id'):
+            self.filtered(lambda o: o.state in ('sent', 'sale')).message_subscribe(partner_ids=[vals.get('partner_id')])
```

Comment:
```suggestion
            self.filtered(lambda so: so.state in ('sent', 'sale')).message_subscribe(
                partner_ids=[vals['partner_id']]
            )
```
Little nitpicking changes:
* line length > 100
* prefer "clear" variable names (o -> so) to make clear what we are working with
* `.get(key)` unnecessary if you are already sure the key is in the dict, you can access it directly

Reply:
Done